kratos/modeler/mdpa_io_modeler.h

```diff
@@ -0,0 +1,153 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//
+//  Main authors:    Miguel Maso Sotomayor
+//
+
+
+#ifndef KRATOS_MDPA_IO_MODELER_H_INCLUDED
+#define KRATOS_MDPA_IO_MODELER_H_INCLUDED
+
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "modeler.h"
+
+namespace Kratos
+{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @class MdpaIoModeler
+ * @ingroup KratosCore
+ * @brief Mdpa file reading
+ * @details Import a model part from an mdpa file
+ * @author Miguel Maso Sotomayor
+ */
+class KRATOS_API(KRATOS_CORE) MdpaIoModeler : public Modeler
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of Modeler
+    KRATOS_CLASS_POINTER_DEFINITION(MdpaIoModeler);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Default constructor
+     */
+    MdpaIoModeler() : Modeler()
+    {
+    }
+
+    /**
+     * @brief Constructor with Model and Parameters
+     */
+    MdpaIoModeler(Model& rModel, Parameters ModelerParameters = Parameters());
+
+    /**
+     * @brief Destructor
+     */
+    virtual ~MdpaIoModeler() = default;
+
+    /**
+     * @brief Creates the Modeler Pointer
+     */
+    Modeler::Pointer Create(Model& rModel, const Parameters ModelParameters) const override
+    {
+        return Kratos::make_shared<MdpaIoModeler>(rModel, ModelParameters);
+    }
+
+    ///@}
+    ///@name Stages
+    ///@{
+
+    void SetupGeometryModel() override;
```

Comment:
```suggestion
    void SetupModelPart() override;
```

using the SetupModelPart would be more appropriate, here.